[ctzeyong] iP

ctzeyong opened the pull request.

ctzeyong committed:
Level-0

ctzeyong committed:
Rename, Greet, Exit

ctzeyong committed:
Level-1

Echo user input
Exits when command "bye"

ctzeyong committed:
List function

Abstract some lines into methods
Takes input into a list
"list" command displays the full list

ctzeyong committed:
Move list functions to a new TaskList class

ctzeyong committed:
Move list functions to new TaskList class

ctzeyong committed:
Make new Task class to manage done tasks
Rename Boaz to Melchizedek

ctzeyong committed:
Split into smaller methods

ctzeyong committed:
Rename some methods and variables

Chinorea commented on src/main/java/Melchizedek.java:
```diff
@@ -0,0 +1,44 @@
+
+    public static TaskList taskList = new TaskList();
+
+    public static void printHorizontalLine() {
```
You could improve the naming of the function to something more specific, in the event you change the functionality of this.

You can consider renaming it to printParition or printBorder, in the event the border is change to vertical line or a line of + signs

Chinorea commented on src/main/java/Melchizedek.java:
```diff
@@ -0,0 +1,44 @@
+        printHorizontalLine();
+    }
+
+    public static void sayByeToUser() {
```
Good naming convention of methods (), function written in camelCase and is in verb form

Chinorea commented on src/main/java/Melchizedek.java:
```diff
@@ -0,0 +1,44 @@
+    public static TaskList taskList = new TaskList();
+
+    public static void printHorizontalLine() {
+        System.out.println("\t____________________________________________________________");
```
Consider declaring as a constant by using public static final String :.....

Chinorea commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,51 @@
+    public TaskList(String task){
+        allTasks = new Task[100];
+        allTasks[0] = new Task(1, task);
+        noOfTasks = 1;
```
Consider renaming the variable to something more intuitive such as numberOfTasks instead of abreviating number as no

ctzeyong committed:
Create new classes Todo, Event and Deadline, all as subclass of Task

Create override for toString() method for classes Task, Todo, Event and Deadline

ctzeyong committed:
Refactor input processing from if-else statements to switch-case statements

ctzeyong committed:
Add feature to process todo tasks and print accurately

ctzeyong committed:
Add feature to process deadlines and events

ctzeyong committed:
Create automated text UI testing

Fix a bug that causes extra indentation when displaying unmark command

ctzeyong committed:
More readable code

Extract some methods
Add blank lines to structure code logically

ctzeyong committed:
Make Task an abstract class; all tasks must be todo, deadline or event

ctzeyong committed:
Catch exceptions for mark and unmark commands

ctzeyong committed:
Handle exceptions for task inputs

ctzeyong committed:
Clean up unused methods

ctzeyong committed:
Revert "Catch exceptions for mark and unmark commands"

This reverts commit 3a495d60afe8507c9d71d7ee572d192070397251.

ctzeyong committed:
Merge branch 'branch-Level-5'

ctzeyong committed:
Organize into Packages

Update runtest.bat directories

ctzeyong committed:
Merge branch 'branch-A-Packages'

ctzeyong committed:
Change allTasks to ArrayList

Remove taskCount variable from TaskList class
Remove id variable from Task class

ctzeyong committed:
add support for deleting tasks

update automatic text-ui-test

ctzeyong committed:
Add method to create and locate new file to store data

ctzeyong committed:
Create method to convert task to file saving format

ctzeyong committed:
Merge branch 'branch-Level-6'

ctzeyong committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/melchizedek/task/TaskList.java

ctzeyong committed:
Debug code left from merge

Introduce more exception handling

ctzeyong committed:
Fix issue with loading previous sessions

ctzeyong committed:
Create new package for exceptions

ctzeyong committed:
Clean up unused code

Refactor exception handling

thomasjlalba commented on src/main/java/melchizedek/Melchizedek.java:
```diff
@@ -0,0 +1,150 @@
+        sayHelloToUser();
+        FileHandling.loadFile(taskList);
+        Scanner in = new Scanner(System.in);
+        while (true) {
```
Maybe the looping logic could be extracted out to avoid long methods?

thomasjlalba commented on src/main/java/melchizedek/Melchizedek.java:
```diff
@@ -0,0 +1,150 @@
+
+            printSeparator();
+
+            switch (tokens[0].toLowerCase()) {
```
Maybe `tokens[0]` could be refactored to make it clearer that it is the command? Same for `tokens[1]` and a few others!

thomasjlalba commented on src/main/java/melchizedek/Melchizedek.java:
```diff
@@ -0,0 +1,150 @@
+                    if (tokens.length < 2) {
+                        throw new DescriptionNotPresentException();
+                    }
+                    taskList.addEvent(Arrays.copyOfRange(tokens, 1, tokens.length));
```
Maybe it would be good to refactor this magic literal `1` that occurs in several places!

thomasjlalba reviewed (commented):
Some refactoring could be done on magic literals and long methods, but well done on following the coding standards and quality.

ctzeyong committed:
Create new classes Ui, Parser

Rename FileHandling class to Storage

ctzeyong committed:
Add method to Parser class

ctzeyong committed:
Reorganise classes and methods

ctzeyong committed:
Implement find feature

ctzeyong committed:
Merge pull request #1 from ctzeyong/branch-Level-9

Implement find feature

ctzeyong committed:
Merge branch 'master' into branch-A-JavaDoc

ctzeyong committed:
Add JavaDoc to most methods and classes

ctzeyong committed:
Merge pull request #2 from ctzeyong/branch-A-JavaDoc

Add JavaDoc

ctzeyong committed:
Merge branch 'master' into branch-Level-8

ctzeyong committed:
Start on user guide

ctzeyong committed:
Update user guide to include all commands

ctzeyong committed:
Update user guide again

ctzeyong committed:
Merge branch 'master' into branch-Level-8

ctzeyong committed:
Add date storing for deadlines

ctzeyong committed:
Merge pull request #3 from ctzeyong/branch-Level-8

Branch level 8

ctzeyong committed:
Handle exception for LocalDate parsing

Add feature for LocalTime for deadlines

ctzeyong committed:
Add feature to handle Date and Time for Event class

ctzeyong committed:
Clean up unused code and update JavaDoc

ctzeyong committed:
Update README
